linkerd/app/inbound/src/http/mod.rs

```diff
@@ -238,3 +238,127 @@ fn trace_labels() -> std::collections::HashMap<String, String> {
     l.insert("direction".to_string(), "inbound".to_string());
     l
 }
+
+#[cfg(fuzzing)]
+pub mod fuzz_logic {
+    use crate::{
+        target::{HttpAccept, TcpAccept, TcpEndpoint},
+        test_util::{
+            support::{connect::Connect, http_util, profile, resolver},
+            *,
+        },
+        Config, Inbound,
+    };
+    use hyper::http;
+    use hyper::{client::conn::Builder as ClientBuilder, Body, Request, Response};
+    use linkerd_app_core::{
+        io::{self, BoxedIo},
+        proxy,
+        svc::{self, NewService, Param},
+        tls,
+        transport::{ClientAddr, Remote, ServerAddr},
+        Conditional, Error, NameAddr, ProxyRuntime,
+    };
+    pub use linkerd_app_test as support;
+    use linkerd_app_test::*;
+
+    pub async fn fuzz_entry_raw(uri: &str, header_data_1: &str, header_data_2: &str, is_get: bool) {
+        let mut server = hyper::server::conn::Http::new();
+        server.http1_only(true);
+        let mut client = ClientBuilder::new();
+        let accept = HttpAccept {
+            version: proxy::http::Version::Http1,
+            tcp: TcpAccept {
+                target_addr: ([127, 0, 0, 1], 5550).into(),
+                client_addr: Remote(ClientAddr(([10, 0, 0, 41], 6894).into())),
+                tls: Conditional::None(tls::server::NoServerTls::NoClientHello),
+            },
+        };
+        let connect =
+            support::connect().endpoint_fn_boxed(accept.tcp.target_addr, hello_fuzz_server(server));
+        let profiles = profile::resolver();
+        let profile_tx = profiles
+            .profile_tx(NameAddr::from_str_and_port("foo.svc.cluster.local", 5550).unwrap());
+        profile_tx.send(profile::Profile::default()).unwrap();
+
+        // Build the outbound server
+        let cfg = default_config();
+        let (rt, _shutdown) = runtime();
+        let server = build_fuzz_server(cfg, rt, profiles, connect).new_service(accept);
+        let (mut client, bg) = http_util::connect_and_accept(&mut client, server).await;
+
+        let http_method = if is_get {
+            http::Method::GET
+        } else {
+            http::Method::POST
+        };
+
+        if let Ok(req) = Request::builder()
+            .method(http_method)
+            .uri(uri)
+            .header(header_data_1, header_data_2)
+            .body(Body::default())
+        {
+            let rsp = http_util::http_request(&mut client, req).await;
+            let _body = http_util::body_to_string(rsp.into_body()).await;
+
+            // Let's send one with a correct URL
+            let req2 = Request::builder()
+                .method(http::Method::GET)
+                .uri("http://foo.svc.cluster.local:5550")
+                .body(Body::default())
+                .unwrap();
```

Comment:
just any traffic at this point. I think this fuzzer would benefit from creating a sequence of requests rather than just a single random request (as it is now), however, I would like to test the whole end-to-end concept with a single random request at first.